### PR TITLE
Merging core into igvc

### PR DESCRIPTION
**Pull request by garethellis0**

### CODE REVIEW

**Comment by garethellis0** [comment]
The review process here should be a fairly simple matter of making sure that all the changes being merged over are sane (ie. we want them in `igvc`)

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,3 @@ cmake-build-debug
 *_REMOTE_*.txt
 *.orig
 
-# Symbols database for stuff like autocomplete in Vim
```
Comment (ValRat) [inline .gitignore]: I'm gonna guess this is intentional?

Comment (garethellis0) [reply .gitignore]: Yeah, we now have a symlink there for the `compile_commands.json` file, so we don't want to ignore it anymore.

**Comment by garethellis0** [comment]
@ValRat @raadkhan - ready for re-review

**Pull request merged**